crates/oxc_formatter/src/lib.rs

```diff
@@ -218,3 +252,44 @@ impl Formatter {
         self.code.last()
     }
 }
+
+fn auto_detect_end_of_line(raw_input_text: &str) -> FinalEndOfLine {
```

Comment:
Can we reference the doc on what this does, i.e. from the doc: 

> "auto" - Maintain existing line endings (mixed values within one file are normalised by looking at what’s used after the first line)

Also, does it make sense to put these methods under `EndOfLine`? i.e.

```rust
impl EndOfLine {
 fn auto_detect_end_of_line ...
 fn get_final_end_of_line ...
}